File: htmltmpl-php/test/nestloop.php
<?

$TEST = "nestloop";
require("head.inc");

#######################################################

$tproc->set("title", "Template world.");
$tproc->set("greeting", "Hello !");
$tproc->set("Loop1", array(
    array( "Loop2" => array( array(), array() ), 
           "Loop3" => array(), 
           "Loop4" => array( array( "Loop6" => array(
        array(), array(), array()
      ) ) ), 
           "Loop5" => array() ),

    array( "Loop2" => array(), 
      "Loop3" => array( array(), array() ), 
      "Loop4" => array(), 
      "Loop5" => array( array() ) )
));

#######################################################

require("foot.inc");

?>

File: htmltmpl/test/nestloop.py
#!/usr/bin/env python

TEST = "nestloop"

import sys
import os
sys.path.insert(0, "..")

from htmltmpl import Template

tmpl = Template(template = TEST + ".tmpl",
                template_path = ["."],
                compile = 0,
                debug = "debug" in sys.argv)

#######################################################

tmpl["title"] = "Template world."
tmpl["greeting"] = "Hello !"
tmpl["Loop1"] = [
    { "Loop2" : [ {}, {} ], 
      "Loop3" : [], 
      "Loop4" : [ { "Loop6" : [
        {}, {}, {}
      ] } ], 
      "Loop5" : [] },

    { "Loop2" : [], 
      "Loop3" : [ {}, {} ], 
      "Loop4" : [], 
      "Loop5" : [ {} ] }
]

#######################################################

output = tmpl.output()

if "out" in sys.argv:
    sys.stdout.write(output)
    sys.exit(0)

res = open("%s.res" % TEST).read()

print TEST, "...",

if output == res:
    print "OK"
else:
    print "FAILED"
    open("%s.fail" % TEST, "w").write(output)